src/tools/perf/lib/libperf.c

```diff
@@ -980,33 +981,43 @@ static void ucp_perf_release_requests_in_progress(ucx_perf_context_t *perf,
     }
 }
 
+static inline void ucp_perf_test_rkey_destroy(ucp_rkey_h rkey)
+{
+    if (NULL != rkey) {
+        ucp_rkey_destroy(rkey);
+    }
+}
+
 static void
 ucp_perf_test_destroy_eps(ucx_perf_context_t *perf, ucp_ep_domain_t ep_domain)
 {
     unsigned thread_count  = perf->params.thread_count;
+    int daemon_mode        = perf->params.ucp.is_daemon_mode;
     unsigned num_in_prog   = 0;
-    ucs_status_ptr_t *reqs = ucs_alloca(thread_count * sizeof(*reqs));
+    ucs_status_ptr_t *reqs = ucs_alloca(2 * thread_count * sizeof(*reqs));
     ucs_status_ptr_t req;
+    ucx_perf_context_t *thread_perf;
     unsigned i;
 
-#define UCP_RKEY_DESTROY(_rkey) \
-    if ((_rkey) != NULL) { \
-        ucp_rkey_destroy(_rkey); \
-    }
-
     for (i = 0; i < thread_count; ++i) {
+        thread_perf = &perf->ucp.tctx[i].perf;
         if (ep_domain & UCP_EP_DOMAIN_SELF) {
-            UCP_RKEY_DESTROY(perf->ucp.tctx[i].perf.ucp.self_send_rkey);
-            UCP_RKEY_DESTROY(perf->ucp.tctx[i].perf.ucp.self_recv_rkey);
+            ucp_perf_test_rkey_destroy(thread_perf->ucp.self_send_rkey);
+            ucp_perf_test_rkey_destroy(thread_perf->ucp.self_recv_rkey);
+
+            if ((req = ucp_perf_test_destroy_ep(thread_perf->ucp.self_ep, i,
```

Comment:
IMO better separate "req = ..." to previous line. same for line 1017

Reply:
ok